### PR TITLE
Port to Pyro

### PR DESCRIPTION
**Pull request by willnewton**

### CODE REVIEW

**Comment by willnewton** [comment]
This also requires this upstream Poky bug to be fixed:

https://bugzilla.yoctoproject.org/show_bug.cgi?id=12045

File: layers/meta-resin-raspberrypi/conf/samples/local.conf.sample [hunk 1]
```diff
@@ -62,3 +62,5 @@ BB_DISKMON_DIRS = "\
     ABORT,${SSTATE_DIR},100M,1K"
 
 CONF_VERSION = "1"
+
+HOSTTOOLS += "docker"
```
Comment (floion) [inline layers/meta-resin-raspberrypi/conf/samples/local.conf.sample]: Can you point to some docs on the usage of this var? Or maybe detail on it in the git log?

Comment (willnewton) [reply layers/meta-resin-raspberrypi/conf/samples/local.conf.sample]: http://www.yoctoproject.org/docs/current/ref-manual/ref-manual.html#var-HOSTTOOLS

It looks like Pyro is strict about what host binaries are allowed to be called from recipes.

**Comment by willnewton** [comment]
@resin-jenkins retest this please

**Comment by willnewton** [comment]
Still required before this is no longer WIP:
 1. meta-resin release
 2. Upstream poky fix for Go binaries

File: layers/meta-resin-raspberrypi/conf/layer.conf [hunk 2]
```diff
@@ -14,32 +14,45 @@ BBFILE_PRIORITY_resin-raspberrypi = "1337"
...
     overlays/dionaudio-loco.dtbo \
+    overlays/dionaudio-loco-v2.dtbo \
     overlays/dpi18.dtbo \
     overlays/dpi24.dtbo \
     overlays/dwc2.dtbo \
     overlays/dwc-otg.dtbo \
     overlays/enc28j60.dtbo \
     overlays/enc28j60-spi2.dtbo \
+    overlays/fe-pi-audio.dtbo \
+    overlays/goodix.dtbo \
```
Comment (agherzan) [inline layers/meta-resin-raspberrypi/conf/layer.conf]: Very nice!

**Comment by willnewton** [comment]
The current status of this PR:
 1. One more meta-resin release is required (to include c7677d8548ce0f2192e15584e3a0c1ae00bb72a2)
 1. Get goarch.bbclass fix merged to poky upstream pyro branch (already accepted for master)

**Comment by willnewton** [comment]
Current status: still waiting for upstream fix for GOARM setting. It has been accepted for the pyro branch but not yet applied to it.

**Comment by floion** [comment]
@willnewton any blockers left for this PR to get merged?

**Comment by willnewton** [comment]
@floion its still blocked on fixing ```goarch.bbclass``` to get the value of ```GOARM``` set correctly. Merged as is it would work for rpi3 but break on rpi (as Go binaries would be built for ARMv7 on all platforms).

The two solutions I am aware of are:
 1. Wait for upstream to merge the fix to the pyro branch of poky (it will get there but no idea when)
 1. I copied a fixed ```goarch.bbclass``` to ```meta-resin/meta-resin-pyro``` but that doesn't work. Classes are looked up in the order of ```BBLAYERS``` which is currently alphabetically ordered so we cannot override the classes in poky with classes in meta-resin unless we reorder the list in ```bblayers.conf.sample```. We could do that as a short term hack until the upstream commit is merged but it feels ugly.

**Comment by shaunmulligan** [comment]
@floion or @agherzan do either of you have a better solution that .2) that @willnewton proposed? would be good to get the kernel and firmware changes out to users

**Comment by floion** [comment]
I would like for #1 to be done. Can we re-ping the yocto guys and ask for a probable date when this gets in @willnewton?

**Comment by agherzan** [comment]
@willnewton please give us an update on what happened with the poky patch, please.

**Comment by willnewton** [comment]
@agherzan The required patch was merged to the upstream pyro branch here: http://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/?h=pyro&id=50cc15335ebcdfa2a2fc52c11fa01695dfdb9e26

**Comment by agherzan** [comment]
That brings me joy and happiness. Both in the same time.

**Pull request merged**